### PR TITLE
compress function for numpy

### PR DESCRIPTION
**Pull request by dash96**
#14179

### CODE REVIEW

**Comment by dash96** [comment]
I am using this algorithm to build this function

Step 1) Use ivy.split( ) to split the matrix in the desired form
Step 2) Then use enumerate( ) function to select only those arrays for which the condition is true
Step 3) Use ivy.concat( ) to combine those arrays to give the final desired output

**Comment by dash96** [comment]
Can you please help me understand where I am going wrong. Because I guess I have done a mistake in importing examples in the test function, but not sure how to fix it.

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_indexing_routines/test_indexing_like_operations.py [hunk 1]
```diff
@@ -272,3 +272,33 @@ def test_numpy_fill_diagonal(
...
+    test_flags,
+):
+    input_dtype, value = dtype_value
+    helpers.test_frontend_function(
+        input_dtypes=input_dtype,
+        frontend=frontend,
+        test_flags=test_flags,
+        fn_tree=fn_tree,
+        on_device=on_device,
+        ary=value[0],
```
Comment (valeriocost) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_indexing_routines/test_indexing_like_operations.py:302]: I think you misspelled `arr` here

**Comment by valeriocost** [comment]
@dash96 hey, try with this fix 😄

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by dash96** [comment]
I am very sorry for not being able to focus on this task. I had exams followed by an emergency so had to travel home to India. If someone help me understand the problem, I am facing with this task I would love to fix it and finish the task.

**Pull request closed without merging**